src/admin/index.js

```diff
@@ -190,15 +190,36 @@ module.exports = ({
         await cluster.refreshMetadataIfNecessary()
 
         const metadata = cluster.findTopicPartitionMetadata(topic)
-        const offsets = await cluster.fetchTopicsOffset([
+        const high = await cluster.fetchTopicsOffset([
           {
             topic,
+            fromBeginning: false,
             partitions: metadata.map(p => ({ partition: p.partitionId })),
           },
         ])
 
-        const { partitions } = offsets.pop()
-        return partitions.map(({ partition, offset }) => ({ partition, offset }))
+        const low = await cluster.fetchTopicsOffset([
+          {
+            topic,
+            fromBeginning: true,
+            partitions: metadata.map(p => ({ partition: p.partitionId })),
+          },
+        ])
+
+        const { partitions } = high.pop()
+        const offsets = partitions.map(({ partition, offset }) => ({
+          partition,
+          offset,
+          high: offset,
+        }))
+
+        for (const { partition, offset } of low.pop().partitions) {
```

Comment:
Took me a minute to understand that this was mutating `offsets`. How about moving this into the map above instead? Something like:

```javascript
const { partitions: lowPartitions } = low.pop()
const offsets = partitions.map(({ partition, offset }) => ({
  partition,
  offset,
  high: offset,
  low: lowPartitions.find(({ partition: lowPartition }) => lowPartition === partition).offset
}))
```

Reply:
good point, fixed